File: frontend/src/app/model/notifications/notification.ts
import { Car } from "../car/car";
import { Driver } from "../driver/driver";
import {NotificationStatus} from "./notificationStatus";

export class NotificationM {
  id?:number;
  userEmail: string;
  message: string;
  startLocation?: string;
  endLocation?: string;
  intermediateLocations?: string[];
  duration?: number;
  price?: number;
  reservationTime?: string;
  notificationStatus: NotificationStatus;
  creationDate: string;
  driverEmail?:string;
  drivingId?:number;
  driverData: Driver;
  carData: Car;
  driverStr: string;
  carStr: string;
}

File: backend/src/main/java/com/izzydrive/backend/model/Notification.java
package com.izzydrive.backend.model;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class Notification {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private double price;

    @Column
    private LocalDateTime reservationDate;

    @Column
    private String message;

    @Column
    private double duration;

    @Column
    private double distance;

    @Column
    private String startLocation;

    @Column
    private String endLocation;

    @Enumerated(EnumType.STRING)
    private NotificationStatus notificationStatus;

    @Column
    private String userEmail;

    @Column
    private LocalDateTime creationDate;

    @Column
    private String driverEmail;

    @Column
    private Long drivingId;

    @Column
    private String drivarStr;

    @Column
    private String carStr;
}
